lib/hackmode/realmode.js

```diff
@@ -791,65 +821,70 @@ class N3hRealMode extends N3hMode {
-      throw new Error('Unknown / untracked DNA:' + dnaAddress)
+    if (!(chainId in this._dhtPerChain)) {
+      log.w('Unknown or untracked chain: ' + chainId)
+      return
     }
-    return this._dhtPerDna[dnaAddress]
+    return this._dhtPerChain[chainId]
   }
 
   /**
    * @private
    */
-  _getMemRef (dnaAddress) {
-    if (!(dnaAddress in this._memory)) {
+  _getMemRef (chainId) {
+    if (!(chainId in this._memory)) {
       const mem = new RealMem()
       // send IPC handleStoreEntry on inserting an entryAspect
       mem.registerIndexer((store, data) => {
         if (data && data.type === 'entryAspect') {
           log.t(this.me() + ' got entryAspect', data)
 
-          const encodedAspect = Buffer.from(data.aspect).toString('base64')
-          this._getDhtRef(dnaAddress).post(DhtEvent.dataHoldRequest(data.entryAddress, [encodedAspect]))
+          const encodedAspect = Buffer.from(JSON.stringify(data.aspect)).toString('base64')
+          this._getDhtRef(chainId).post(DhtEvent.dataHoldRequest(data.entryAddress, [encodedAspect]))
 
           // Store in Core if it isn't already holding it
-          if (this._hasEntryAspect(dnaAddress, data.entryAddress, data.address)) {
+          if (this._hasEntryAspect(chainId, data.entryAddress, data.address)) {
             log.t(this.me() + ' entryAspect is known:', data.address)
             return
           }
           // log.t(this.me() + ' dhtEntry is unknown', data.address)
           // bookkeep
-          const chainId = this._intoChainId(dnaAddress, data.providerAgentId)
+          // const chainId = this._intoChainId(dnaAddress, data.providerAgentId)
           this._bookkeepAspect(this._storedEntryBook, chainId, data.entryAddress, data.address)
-          // log.t(this.me() + ' Sending IPC handleStoreEntry: ', data.address)
+          log.t(this.me() + ' Sending IPC handleStoreEntryAspect: ', data.entryAddress, data.aspect)
           this._ipcSend('json', {
             method: 'handleStoreEntryAspect',
             _id: this._generateRequestId(),
-            dnaAddress,
+            dnaAddress: this._deconstructChainId(chainId)[0],
             providerAgentId: data.providerAgentId,
             address: data.entryAddress,
             entryAspect: data.aspect
           })
         }
       })
-      this._memory[dnaAddress] = mem
+      this._memory[chainId] = mem
     }
-    return this._memory[dnaAddress]
+    return this._memory[chainId]
   }
 
   /**
    *  Check if agent is tracking dna.
    *  If not, will try to send a FailureResult back to sender (if sender info is provided).
    *  Returns peerAddress of receiverAgentId if agent is tracking dna.
    *  @private
+   *  FIXME
    */
-  _getPeerAddressOrFail (dnaAddress, receiverAgentId, senderAgentId, requestId) {
-    const recvPeer = this._getDhtRef(dnaAddress).getPeerLocal(receiverAgentId)
-    if (!recvPeer) {
+  _getPeerAddressOrFail (dnaAddress, agentId, receiverAgentId, senderAgentId, requestId) {
```

Comment:
suggest changing `agentId` to `chainAgentId`

Reply:
or perhaps `providerAgentId`?

Reply:
outdated code